nullaway/src/main/java/com/uber/nullaway/handlers/contract/ContractHandler.java

```diff
@@ -203,17 +205,17 @@ public NullnessHint onDataflowVisitMethodInvocation(
     return NullnessHint.UNKNOWN;
   }
 
-  private void reportMatch(Tree errorLocTree, String message) {
-    assert this.analysis != null && this.state != null;
-    if (this.analysis != null && this.state != null) {
-      this.state.reportMatch(
+  protected static void reportMatch(
```

Comment:
It is error prone to be overriding static methods in this way.  If you need to do method overriding, best to just make the method an instance method?  Or use different names?